### PR TITLE
[Relay][Onnx][Frontend] Add RandomUniform converter and tests to onnx frontend.

### PR DESCRIPTION
**Pull request by jwfromm**
This PR adds a converter for the Onnx RandomUniform operator using the recently added PRNG kernels. Note that since it's random, it's difficult to compare correctness to onnxruntime. Instead I confirm basic properties of the produced values. Please let me know if there are better tests that we could perform.

### CODE REVIEW

**Review by mbrookhart — APPROVED** [review]

File: tests/python/frontend/onnx/test_forward.py [hunk 1]
```diff
@@ -4872,6 +4872,48 @@ def test_qlinearadd():
...
+        [node],
+        "random_uniform_test",
+        inputs=[],
+        outputs=[helper.make_tensor_value_info("out", ONNX_DTYPE, shape)],
+    )
+    model = helper.make_model(graph, producer_name="random_uniform_test")
+    return get_tvm_output_with_vm(model, [], target=target, device=tvm.device(target, 0))
+
+
+def test_random_uniform():
```
Comment (AndrewZhaoLuo) [inline tests/python/frontend/onnx/test_forward.py:4894]: Can we also have a test which matches against a golden vector/tensor? 

I think actually testing for a uniform random distribution is really hard but having a golden vector/tensor at least shows to a user things look uniformly random for one seeded value.

**Review by AndrewZhaoLuo — CHANGES_REQUESTED** [review]

File: tests/python/frontend/onnx/test_forward.py (hunk 1 above)
Comment (AndrewZhaoLuo) [reply tests/python/frontend/onnx/test_forward.py:4894]: E.g. https://github.com/numpy/numpy/blob/main/numpy/random/tests/test_random.py#L950

Comment (jwfromm) [reply tests/python/frontend/onnx/test_forward.py:4894]: Yeah thats a great idea, thanks! I've added a test against a golden vector.

**Review by AndrewZhaoLuo — APPROVED** [review]
LGTM

**Comment by jwfromm** [comment]
@mbrookhart it looks like this hit a flaky GPU test for `crop_and_resize`, specifically nans showed up in the output. Is it possible that  could happen due to your recent resize1d changes or did we just get unlucky here?

**Comment by mbrookhart** [comment]
@jwfromm I noticed this on a couple of PRs this morning and tagged you on a fix a bit ago :D https://github.com/apache/tvm/pull/8431

**Comment by jwfromm** [comment]
Oh neat, thanks!

File: tests/python/frontend/onnx/test_forward.py [hunk 2]
```diff
@@ -4872,6 +4872,66 @@ def test_qlinearadd():
...
+        # Test N-D tensor generation.
+        vals = get_random_uniform([1, 3, 100, 100], dtype="float32", target=target)
+        assert list(vals.shape) == [1, 3, 100, 100]
+
+        # Check that bounds aren't exceeded.
+        vals = get_random_uniform(shape=[100], high=100, low=-100)
+        assert list(vals.shape) == [100]
+        assert all(vals >= -100) and all(vals <= 100)
+
+        # Check that fixed seed produces the same values.
```
Comment (emijiayw) [inline tests/python/frontend/onnx/test_forward.py:4911]: Check that if fixed seeds will produce the same values.

**Pull request merged**